core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -95,13 +97,25 @@
     private int maxTasksWaitingForNodePerStage = 5;
 
     private boolean enabledAdaptiveTaskRequestSize = true;
-    private DataSize maxRemoteTaskRequestSize = DataSize.of(8, DataSize.Unit.MEGABYTE);
-    private DataSize remoteTaskRequestSizeHeadroom = DataSize.of(2, DataSize.Unit.MEGABYTE);
+    private DataSize maxRemoteTaskRequestSize = DataSize.of(8, MEGABYTE);
+    private DataSize remoteTaskRequestSizeHeadroom = DataSize.of(2, MEGABYTE);
     private int remoteTaskGuaranteedSplitPerTask = 3;
 
-    private DataSize faultTolerantExecutionTargetTaskInputSize = DataSize.of(4, GIGABYTE);
+    private int faultTolerantExecutionArbitraryDistributionComputeTaskTargetSizeGrowthPeriod = 10;
+    private double faultTolerantExecutionArbitraryDistributionComputeTaskTargetSizeGrowthFactor = 2.0;
+    private DataSize faultTolerantExecutionArbitraryDistributionComputeTaskTargetSizeMin = DataSize.of(512, MEGABYTE);
+    private DataSize faultTolerantExecutionArbitraryDistributionComputeTaskTargetSizeMax = DataSize.of(4, GIGABYTE);
 
-    private int faultTolerantExecutionTargetTaskSplitCount = 64;
+    private int faultTolerantExecutionArbitraryDistributionWriteTaskTargetSizeGrowthPeriod = 15;
+    private double faultTolerantExecutionArbitraryDistributionWriteTaskTargetSizeGrowthFactor = 1.5;
+    private DataSize faultTolerantExecutionArbitraryDistributionWriteTaskTargetSizeMin = DataSize.of(4, GIGABYTE);
+    private DataSize faultTolerantExecutionArbitraryDistributionWriteTaskTargetSizeMax = DataSize.of(16, GIGABYTE);
+
+    private DataSize faultTolerantExecutionHashDistributionComputeTaskTargetSize = DataSize.of(512, MEGABYTE);
+    private DataSize faultTolerantExecutionHashDistributionWriteTaskTargetSize = DataSize.of(4, GIGABYTE);
+    private int faultTolerantExecutionHashDistributionWriteTaskCountSoftUpperBound = 3000;
```

Comment:
I do not have a good gut feeling here. Is 3k good value @arhimondr ?

Reply:
Updated to 2000